### PR TITLE
feat(jira): Add project status mapping to org integration config

### PR DESCRIPTION
**Pull request by macqueen**
see
https://github.com/getsentry/sentry/pull/9163

### CODE REVIEW

File: src/sentry/integrations/jira/integration.py [hunk 1]
```diff
@@ -139,6 +139,28 @@ def get_project_config(self):
...
+        """
+        config = self.org_integration.config
+
+        if 'sync_status_forward' in data:
+            project_mappings = data.pop('sync_status_forward')
+
+            data['sync_status_forward'] = bool(project_mappings)
+
+            for project_id, statuses in project_mappings.items():
+                IntegrationExternalProject.objects.create(
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/jira/integration.py]: What about removing them?

Comment (lauryndbrown) [reply src/sentry/integrations/jira/integration.py]: A test for removal i.e. deleting `IntegrationExternalProjects` that have been removed from the UI  and checking that `'sync_status_forward'` is set to `False` if all of the project mappings are removed would be useful here.

Comment (lauryndbrown) [reply src/sentry/integrations/jira/integration.py]: Also don't forget about just updating pre-existing `IntegrationExternalProjects`

**Review by lauryndbrown — COMMENTED** [review]
I think it looks ok. As long as @EvanPurkhiser is happy. 

I hadn't thought of just deleting and recreating. I did something in mine with exclude and update https://github.com/getsentry/sentry/pull/9176 which while not super complicated is no where near as elegant. Neat!

**Review by evanpurkhiser — APPROVED** [review]
🤷‍♂️ seems fine to me

**Pull request merged**